packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -355,35 +356,35 @@ class AndroidDevice extends Device {
       printTrace('Waiting for observatory port to be available...');
 
       try {
-        int observatoryDevicePort, diagnosticDevicePort;
+        Uri observatoryDeviceUri, diagnosticDeviceUri;
 
         if (debuggingOptions.buildMode == BuildMode.debug) {
-          Future<List<int>> scrapeServicePorts = Future.wait(
-              <Future<int>>[observatoryDiscovery.nextPort(), diagnosticDiscovery.nextPort()]
+          Future<List<Uri>> scrapeServicePorts = Future.wait(
+              <Future<Uri>>[observatoryDiscovery.nextUri(), diagnosticDiscovery.nextUri()]
           );
-          List<int> devicePorts = await scrapeServicePorts.timeout(new Duration(seconds: 20));
-          observatoryDevicePort = devicePorts[0];
-          diagnosticDevicePort = devicePorts[1];
+          List<Uri> deviceUris = await scrapeServicePorts.timeout(new Duration(seconds: 20));
+          observatoryDeviceUri = deviceUris[0];
+          diagnosticDeviceUri = deviceUris[1];
         } else {
-          observatoryDevicePort = await observatoryDiscovery.nextPort().timeout(new Duration(seconds: 20));
+          observatoryDeviceUri = await observatoryDiscovery.nextUri().timeout(new Duration(seconds: 20));
         }
 
-        printTrace('observatory port on device: $observatoryDevicePort');
+        printTrace('observatory URI on device: $observatoryDeviceUri');
```

Comment:
AndroidDevice:
printTrace('observatory URI on device:' ...)

IosDevice:
printTrace("Local Observatory: ${uris[0]}");

IosSimulator:
        printTrace('service protocol = $deviceUri');


Please pick one message and be consistent across all devices.